### DISCUSSION TITLE
Map differend ldap-groups

### DISCUSSION THREAD

**Discussion by xundeenergie** [sventorben/keycloak-restrict-client-auth#251] [Q&A] [answered]
I've configured in my ldap (where the users and groups/roles come from in my ldap) for each application/client roles with such a pattern:
perm-app-<appname>_users
perm-app-<appname>_admins

And i've a mapper in keycloak for those ldap-groups to create and assign them as roles for my users. 

How can i map such roles with patterns to the client-role "restricted-access"?

Or can i configure the access-provider with a wildcard/pattern?
Is it possible with something like
`spi-restrict-client-auth-access-provider-client-role-client-role-name=perm-app-${clientid}_*`

**Comment by sventorben** [comment]
Are your roles `perm-app-_users` and `perm-app-_admins` per client or are these realm roles?

**Reply by sventorben** [reply, marked as answer]
Forget about my question. Try the following:

Add a client role "restricted-access" to each app/client. Then convert your `perm-app-_users` and `perm-app-_admins` roles to composite roles that include the `restricted-access` role of the corresponding app/client.

See video vor details:

https://github.com/sventorben/keycloak-restrict-client-auth/assets/12183470/b0b6064a-9524-4bae-9024-0347c47c3071

**Reply by sventorben** [reply]
@xundeenergie does this work for you?

**Reply by xundeenergie** [reply]
yes, it works for me.
Thank you!!
